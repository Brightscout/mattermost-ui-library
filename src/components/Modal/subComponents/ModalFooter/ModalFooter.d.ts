import React from 'react';

export interface ModalFooterProps {

    /**
     * Primary action component
     */
    primary: React.ReactNode;

    /**
     * Secondary action component
     */
    secondary?: React.ReactNode;
}