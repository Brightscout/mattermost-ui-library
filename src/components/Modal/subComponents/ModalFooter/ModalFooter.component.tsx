import React from 'react';

import {ModalFooterProps} from './ModalFooter';
import {ModalFooterWrapper} from './ModalFooter.styles';

/**
 * ModalFooter component
 *
 * @example Correct usage
 * <ModalFooter
 *      primary={<Button variant='primary'>{'Submit'}</Button>}
 *      secondary={<Button variant='tertiary'>{'Close'}</Button>}
 * />
 */
export const ModalFooter = (props: ModalFooterProps) => {
    const {primary, secondary} = props;
    return (
        <ModalFooterWrapper className='mm=modalFooter'>
            {secondary}
            {primary}
        </ModalFooterWrapper>);
};