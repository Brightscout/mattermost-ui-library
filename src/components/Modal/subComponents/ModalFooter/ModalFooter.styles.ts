import styled from 'styled-components';

import Colors from '@Styles/colorsForJs.module.scss';

export const ModalFooterWrapper = styled.div({
    display: 'flex',
    justifyContent: 'flex-end',
    gap: '8px',
    padding: '24px 32px',
    borderTop: `1px solid ${Colors.centerChannel_16}`,
});