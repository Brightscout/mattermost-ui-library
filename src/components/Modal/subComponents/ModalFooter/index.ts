export {ModalFooter} from './ModalFooter.component';
export type {ModalFooterProps} from './ModalFooter';
