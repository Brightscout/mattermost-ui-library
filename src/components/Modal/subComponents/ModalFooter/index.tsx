import React from 'react';
import {Modal as RBModal} from 'react-bootstrap';

type ModalFooterProps = {
    onConfirm?: (() => void) | null;
    confirmBtnText?: string;
    cancelBtnText?: string;
    onHide?: () => void;
    className?: string;
    confirmDisabled?: boolean;
    cancelDisabled?: boolean;
    confirmBtnClassName?: string;
}

const ModalFooter = ({
    onConfirm,
    onHide,
    cancelBtnText,
    confirmBtnText,
    className = '',
    confirmDisabled,
    cancelDisabled,
    confirmBtnClassName = '',
}: ModalFooterProps) : JSX.Element => (
    <RBModal.Footer className={`modal__footer d-flex flex-row-reverse align-items-center ${className}`}>
        {onConfirm && (
            <button
                className={`btn modal__confirm-btn margin-left-10 ${confirmBtnClassName || 'btn-primary'}`}
                onClick={onConfirm}
                disabled={confirmDisabled}
            >
                {confirmBtnText || 'Confirm'}
            </button>
        )}
        {onHide && (
            <button
                className='btn btn-link modal__cancel-btn margin-0 padding-0'
                onClick={onHide}
                disabled={cancelDisabled}
            >
                {cancelBtnText || 'Cancel'}
            </button>
        )}
    </RBModal.Footer>
);

export default ModalFooter;
