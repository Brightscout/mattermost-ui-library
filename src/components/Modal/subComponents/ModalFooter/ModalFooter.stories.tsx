import React from 'react';
import {Story, Meta} from '@storybook/react';

import {ModalFooterProps} from './ModalFooter';
import {ModalFooter} from './ModalFooter.component';

export default {
    title: 'Component/Modal/ModalFooter',
    component: ModalFooter,
} as Meta<ModalFooterProps>;

const ModalFooterTemplate: Story<ModalFooterProps> = (args) => <ModalFooter {...args}/>;

// Default
export const Default = ModalFooterTemplate.bind({});
