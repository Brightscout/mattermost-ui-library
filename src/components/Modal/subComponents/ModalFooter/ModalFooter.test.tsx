import React from 'react';
import {mount, ReactWrapper} from 'enzyme';

import ModalFooter from '.';

// Mock props data
const confirmHandler = jest.fn();
const hideHandler = jest.fn();
const confirmBtnText = 'Confirm Btn';
const cancelBtnText = 'Cancel Btn';
const modalFooterProps = {
    onConfirm: confirmHandler,
    onHide: hideHandler,
    className: 'modal-footer-class',
    confirmDisabled: false,
    cancelDisabled: false,
    confirmBtnClassName: 'confirm-btn-class',
};

describe('Modal Footer', () => {
    let component: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = mount(<ModalFooter {...modalFooterProps}/>);
    });

    afterEach(() => {
        component.unmount();
        jest.clearAllMocks();
    });

    it('Should render correctly and matches snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should apply the passed className prop', () => {
        expect(component.hasClass(modalFooterProps.className)).toBeTruthy();
    });

    it('Should render the confirm button correctly with default text', () => {
        expect(component.find('.modal__confirm-btn').text().includes('Confirm')).toBeTruthy();
    });

    it('Should render the cancel button correctly with default text', () => {
        expect(component.find('.modal__cancel-btn').text().includes('Cancel')).toBeTruthy();
    });

    it('Should apply the class to confirm button', () => {
        expect(component.find('.modal__confirm-btn').hasClass(modalFooterProps.confirmBtnClassName)).toBeTruthy();
    });

    it('Should call the onClick handler when the confirm button is clicked', () => {
        component.find('.modal__confirm-btn').simulate('click');
        expect(confirmHandler).toHaveBeenCalledTimes(1);
    });

    it('Should call the onClick handler when the cancel button is clicked', () => {
        component.find('.modal__cancel-btn').simulate('click');
        expect(hideHandler).toHaveBeenCalledTimes(1);
    });
});

describe('Modal Footer - disabled footer buttons', () => {
    let component: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = mount(
            <ModalFooter
                {...modalFooterProps}
                cancelDisabled={true}
                confirmDisabled={true}
            />,
        );
    });

    afterEach(() => {
        component.unmount();
        jest.clearAllMocks();
    });

    it('Shouldn\'t call the onClick handler when the confirm button is clicked', () => {
        component.find('.modal__confirm-btn').simulate('click');
        expect(confirmHandler).toHaveBeenCalledTimes(0);
    });

    it('Shouldn\'t call the onClick handler when the cancel button is clicked', () => {
        component.find('.modal__cancel-btn').simulate('click');
        expect(hideHandler).toHaveBeenCalledTimes(0);
    });
});

describe('Modal Footer - skip rendering the buttons', () => {
    let component: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = mount(<ModalFooter/>);
    });

    afterEach(() => {
        component.unmount();
        jest.clearAllMocks();
    });

    it('Shouldn\'t render the confirm button when onConfirm prop isn\'t passed', () => {
        expect(component.find('.modal__confirm-btn').exists()).toBeFalsy();
    });

    it('Shouldn\'t render the cancel button when onHide prop isn\'t passed', () => {
        expect(component.find('.modal__cancel-btn').exists()).toBeFalsy();
    });
});

describe('Modal Footer - correctly render the passed button text props for confirm and cancel buttons', () => {
    let component: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = mount(
            <ModalFooter
                {...modalFooterProps}
                confirmBtnText={confirmBtnText}
                cancelBtnText={cancelBtnText}
            />,
        );
    });

    afterEach(() => {
        component.unmount();
        jest.clearAllMocks();
    });

    it('Should render the confirm button text correctly', () => {
        expect(component.find('.modal__confirm-btn').text().includes(confirmBtnText)).toBeTruthy();
    });

    it('Should render the cancel button text correctly', () => {
        expect(component.find('.modal__cancel-btn').text().includes(cancelBtnText)).toBeTruthy();
    });
});
