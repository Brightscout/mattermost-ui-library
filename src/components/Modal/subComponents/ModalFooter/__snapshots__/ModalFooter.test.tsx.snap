// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Footer Should render correctly and matches snapshot 1`] = `ReactWrapper {}`;
